### PR TITLE
User hero disabled HoC fallback text

### PR DESCRIPTION
**Pull request by wjordan**
Text block to be displayed if user hero (the block that displays the course the logged-in user is currently working on) needs to be disabled by Gatekeeper during Hour of Code, to minimize database load:

screenshot at 2016-10-27 10 55 52

### CODE REVIEW

File: dashboard/app/views/api/user_hero.html.haml [hunk 1]
```diff
@@ -43,5 +43,62 @@
...
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD1'))} CS in Science Part 1
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD2'))} CS in Science Part 2
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD3'))} CS in Science Part 3
+            %li
+              %a{href: 'https://studio.code.org/s/sciencePD4'} CS in Science Part 4
+        %ul
+          %li Computer Science Fundamentals Online Professional Learning
```
Comment (tanyaparker) [inline dashboard/app/views/api/user_hero.html.haml]: Do we need CSF bullet if there are no courses?

File: dashboard/app/views/api/user_hero.html.haml [hunk 2]
```diff
@@ -7,21 +7,21 @@
...
-                %button.primary{class: 'teacherdashboard', style: "margin-left: 20px;"}= t('see_your_learning_plan')
     - if Gatekeeper.allows('logged_in_studio_hero', default: true)
+      - plc_enrollments = Plc::UserCourseEnrollment.where(user: current_user)
+      - plc_enrollments.each do |course_enrollment|
+        - uncompleted_units = course_enrollment.plc_unit_assignments.where.not(status: [Plc::EnrollmentUnitAssignment::COMPLETED, Plc::EnrollmentUnitAssignment::START_BLOCKED])
+        - if uncompleted_units.any?
+          - unit_enrollment = uncompleted_units.first
+          #plc_course_progress{class: role_class}
+            %h4
+              = t('your_professional_learning_plan')
```
Comment (tanyaparker) [inline dashboard/app/views/api/user_hero.html.haml:18]: Might as well clean up this to be one line? `%h4= t('your_professional_learning_plan')`

File: dashboard/app/views/api/user_hero.html.haml [hunk 3]
```diff
@@ -43,5 +43,62 @@
             #suggestcourse.smalltext= t('home.choose_next_course')
           - else
             #suggestcourse.smalltext= link_to t('home.no_primary_course'), root_url
-
+    - else
+      #body{class: "hero_disabled"}
+        During Hour of Code, we will not be showing the personalized links to the courses you are currently working on.
```
Comment (tanyaparker) [inline dashboard/app/views/api/user_hero.html.haml]: We try to keep Code Studio fully localized. Do you want to or would you mind if I committed to your PR with these localized?

File: dashboard/app/views/api/user_hero.html.haml [hunk 4]
```diff
@@ -43,5 +43,62 @@
...
+          %li Computer Science in Science Online Professional Learning
+          %ul
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD1'))} CS in Science Part 1
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD2'))} CS in Science Part 2
+            %li
+              %a{href: script_path(Script.get_from_cache('sciencePD3'))} CS in Science Part 3
+            %li
+              %a{href: 'https://studio.code.org/s/sciencePD4'} CS in Science Part 4
```
Comment (tanyaparker) [inline dashboard/app/views/api/user_hero.html.haml]: How come some of these are direct studio links and others are `get_from_cache`?

**Comment by tanyaparker** [comment]
This user hero also shows up for signed in users on code.org homepage. I'm blocked running some scripts on my local right now, could you provide a screenshot of what that looks like? Or I'll check in a bit since you're probably offline for the weekend. Are we ok with a super large block on the homepage? Or could we instead put all this information on a zendesk FAQ and link to it? And that would enable us to edit that immediately/frequently by a non-engineer, but we would lose localization. Since it's only going to be up for such a short period of time (and probably won't be the same text for last year) maybe it doesn't _really_ need localization.

File: lib/cdo/script_config.rb [hunk 5]
```diff
@@ -16,4 +16,19 @@ def self.allows_public_caching_for_script(script_name)
...
+    cached_scripts + %w(
+      20-hour
+      course1
+      course2
+      course3
+      course4
+      playlab
+      artist
+      infinity
+      iceage
```
Comment (tanyaparker) [inline lib/cdo/script_config.rb:31]: Are we meant to include all Hour of Code scripts? What about mc, minecraft, starwars, starwarsblocks, frozen, hourofcode, gumball, flappy?

File: dashboard/app/views/api/user_hero.html.haml (hunk 4 above)
Comment (wjordan) [reply dashboard/app/views/api/user_hero.html.haml]: the direct studio links reference courses that do/did not (yet?) exist. they probably need attention.

File: lib/cdo/script_config.rb (hunk 5 above)
Comment (wjordan) [reply lib/cdo/script_config.rb:31]: those are all included in this array via `cached_scripts`. The constant list here only includes scripts that are not already 'cached scripts' (which are all assumed to be 'scale mode') that also need to be marked as 'scale mode' for feature flag management.

**Pull request merged**